x/cdp/keeper/savings.go

```diff
@@ -81,16 +78,14 @@ func (k Keeper) SetPreviousSavingsDistribution(ctx sdk.Context, distTime time.Ti
 	store.Set([]byte{}, k.cdc.MustMarshalBinaryLengthPrefixed(distTime))
 }
 
+// getModuleAccountCoins gets the total coin balance of this coin currently held by module accounts
 func (k Keeper) getModuleAccountCoins(ctx sdk.Context, denom string) sdk.Coins {
-	// NOTE: these are the module accounts that could end up holding stable denoms at some point.
-	// Since there are currently no api methods to 'GetAllModuleAccounts', this function will need to be updated if a
-	// new module account is added which can hold stable denoms.
-	savingsRateMaccCoinAmount := k.supplyKeeper.GetModuleAccount(ctx, types.SavingsRateMacc).GetCoins().AmountOf(denom)
-	cdpMaccCoinAmount := k.supplyKeeper.GetModuleAccount(ctx, types.ModuleName).GetCoins().AmountOf(denom)
-	auctionMaccCoinAmount := k.supplyKeeper.GetModuleAccount(ctx, auctiontypes.ModuleName).GetCoins().AmountOf(denom)
-	liquidatorMaccCoinAmount := k.supplyKeeper.GetModuleAccount(ctx, types.LiquidatorMacc).GetCoins().AmountOf(denom)
-	feeMaccCoinAmount := k.supplyKeeper.GetModuleAccount(ctx, authtypes.FeeCollectorName).GetCoins().AmountOf(denom)
-	kavadistMaccCoinAmount := k.supplyKeeper.GetModuleAccount(ctx, kdtypes.ModuleName).GetCoins().AmountOf(denom)
-	totalModAccountAmount := savingsRateMaccCoinAmount.Add(cdpMaccCoinAmount).Add(auctionMaccCoinAmount).Add(liquidatorMaccCoinAmount).Add(feeMaccCoinAmount).Add(kavadistMaccCoinAmount)
-	return sdk.NewCoins(sdk.NewCoin(denom, totalModAccountAmount))
+	var totalModCoinBalance sdk.Coins
```

Comment:
The way you've written it is safe, but will call `Add` even for module accounts with zero balance. 

Another approach would be: (see next suggestion as well)

```suggestion
	totalModCoinBalance := sdk.NewCoins(sdk.NewCoin(sdk.ZeroInt(), denom))
```

Reply:
https://github.com/Kava-Labs/kava/pull/517/commits/8cdc8c4035bf9cbdba274ba392514b5f271c9f95